lib/puppet/pops/issues.rb

```diff
@@ -545,4 +545,21 @@ def self.hard_issue(issue_code, *args, &block)
   MULTIPLE_ATTRIBUTES_UNFOLD = hard_issue :MULTIPLE_ATTRIBUTES_UNFOLD do
     "Unfolding of attributes from Hash can only be used once per resource body"
   end
+
+  SYNTAX_ERROR = hard_issue :SYNTAX_ERROR, :where do
+    "Syntax error at #{where}"
+  end
+
+  LEX_ERROR = hard_issue :LEX_ERROR do
+    # Error here for completeness. It's never printed
+    "Unable to create lexical token stream"
```

Comment:
Yes, that is much better - and removes the error messages from the source in preparation of I18N. It also provides better feedback in structured output as the code is more meaningful. There are not that many different lex errors, so worth doing.